### PR TITLE
Add a relation from SphinxDomain -> HTMLFile

### PR DESCRIPTION
**Pull request by ericholscher**
This will help with search indexing,
as it will let us get this data through a relation on the model instead of in naive Python code.

Refs #5821

### CODE REVIEW

File: readthedocs/sphinx_domains/admin.py [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 class SphinxDomainAdmin(admin.ModelAdmin):
     list_filter = ('type',)
-    list_display = ('docs_url', 'version', 'build')
-    raw_id_fields = ('project', 'version')
+    list_display = ('__str__', 'version', 'build')
```
Comment (ericholscher) [inline readthedocs/sphinx_domains/admin.py]: This uses _way_ fewer queries than doing a `resolve`

Comment (dojutsu-user) [reply readthedocs/sphinx_domains/admin.py]: How did you actually find that out?

Comment (ericholscher) [reply readthedocs/sphinx_domains/admin.py]: django-debug-toolbar shows the query count 👍

File: readthedocs/sphinx_domains/models.py [hunk 2]
```diff
@@ -32,6 +32,11 @@ class SphinxDomain(TimeStampedModel):
         verbose_name=_('Version'),
         related_name='sphinx_domains',
     )
+    html_file = models.ForeignKey(
+        HTMLFile,
+        related_name='sphinx_domains',
+        null=True,
```
Comment (davidfischer) [inline readthedocs/sphinx_domains/models.py]: I would also suggest `blank=True` unless we're planning to backfill this.

Comment (ericholscher) [reply readthedocs/sphinx_domains/models.py]: The migration just makes it `null`, and it should be filled going forward. Any reason to allow `blank`?

Comment (humitos) [reply readthedocs/sphinx_domains/models.py]: If we want to allow `null` on already existing objects, but force on new ones, it makes sense as it is to me.

**Comment by ericholscher** [comment]
Gonna merge this 🎉

**Pull request merged**